src/components/Identity/createComponent.js

```diff
@@ -78,50 +74,46 @@ export default (idSyncs, config, logger, cookieJar, optIn, eventManager) => {
 
           let promise;
 
-          if (ecid) {
-            addIdsContext(payload, ecid);
-          } else if (deferredForEcid) {
-            // We don't have an ECID, but the first request has gone out to
-            // fetch it. We must wait for the response to come back with the
-            // ECID before we can apply it to this payload.
-            logger.log("Delaying request while retrieving ECID from server.");
-            promise = deferredForEcid.promise.then(() => {
-              logger.log("Resuming previously delayed request.");
-              addIdsContext(payload, getEcid());
-            });
-          } else {
-            // We don't have an ECID and no request has gone out to fetch it.
-            // We won't apply the ECID to this request, but we'll set up a
-            // promise so that future requests can know when the ECID has returned.
-            deferredForEcid = defer();
-            payload.expectResponse();
-            if (
-              config.thirdPartyCookiesEnabled &&
-              areThirdPartyCookiesSupported(getBrowser(window))
-            ) {
-              payload.useIdThirdPartyDomain();
+          if (!ecid) {
+            if (deferredForEcid) {
+              // We don't have an ECID, but the first request has gone out to
+              // fetch it. We must wait for the response to come back with the
+              // ECID before we can apply it to this payload.
+              logger.log("Delaying request while retrieving ECID from server.");
+              promise = deferredForEcid.promise.then(() => {
+                logger.log("Resuming previously delayed request.");
+              });
+            } else {
+              // We don't have an ECID and no request has gone out to fetch it.
+              // We won't apply the ECID to this request, but we'll set up a
+              // promise so that future requests can know when the ECID has returned.
+              deferredForEcid = defer();
+              payload.expectResponse();
+              if (
+                config.thirdPartyCookiesEnabled &&
+                areThirdPartyCookiesSupported(getBrowser(window))
+              ) {
+                payload.useIdThirdPartyDomain();
+              }
             }
           }
+
           customerIds.addToPayload(payload);
           return promise;
         });
       },
 
-      // Waiting for opt-in because we'll be writing the ECID to a cookie
-      onResponse({ response }) {
+      // Waiting for opt-in because we'll be reading the ECID from a cookie
```

Comment:
We should revisit this comment and `onResponse` in general, since we might not read the ECID from a cookie. Maybe.

Reply:
I can see how we could assume we have an identity if we get any response. Are you thinking there's a way to not read the ECID cookie at all though? If so, I'm interested. How would we know whether we need to hit demdex or not? Maybe just look for the existence of any `kndctr_`-prefixed cookie?

Reply:
Until we resolve the whole story of "How does Alloy get the ECID", I think the existence of the `kndctr_org_identity` cookie should be enough to know that you are a returning visitor with an existing ECID.